.gitlab/common/shared.yml

```diff
@@ -35,6 +35,24 @@
   - *get_artifactory_token_win
   - if (($Env:USE_CACHING_PROXY_PYTHON -eq "true") -and ($Env:ARTIFACTORY_BYPASS -eq "false")) { $PIP_INDEX_URL="https://${Env:ARTIFACTORY_USERNAME}:${ARTIFACTORY_TOKEN}@${Env:ARTIFACTORY_URL}/${Env:ARTIFACTORY_PYPI_PATH}" }
 
+# Sets up a cache for gems used by Omnibus
+# Usage:
+# !reference [.cache_omnibus_ruby_deps, setup] somewhere ahead of invoking bundle
+# !reference [.cache_omnibus_ruby_deps, cache] under `cache` for the same job
+.cache_omnibus_ruby_deps:
+  setup:
+    - pushd omnibus && bundle config set --local path 'vendor/bundle' && popd
+  cache:
+    key:
+      files:
+        - omnibus/Gemfile
+        # This is coarser than needed, but there's no more convenient way
+        # to get a hold of OMNIBUS_RUBY_VERSION and OMNIBUS_SOFTWARE version
+        - release.json
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏That's a lot 😭

Reply:
Are you sure the Gemfile is not enough, since it has `ENV['OMNIBUS_RUBY_VERSION']` and `ENV['OMNIBUS_SOFTWARE_VERSION']` in the gems ref ?
<img width="1068" alt="image" src="https://github.com/DataDog/datadog-agent/assets/47357713/2b9aa78e-116b-458b-856b-ed501ee99ec3">

Reply:
We'd need to double check the env always has both those var

Reply:
> I looked at the release.json file's recent history and it didn't seem to me that it was being updated that often

Agreed for `main`, IDK for all branches, but anyway I agree with you to start with simple !